index.js

```diff
@@ -227,6 +227,11 @@ class GithubScm extends Scm {
             privateRepo = scmRepo.privateRepo || false;
         } else {
             try {
+                if (scmHost !== this.config.gheHost) {
```

Comment:
there is a bug if we're using open source (and `gheHost` is not set)

```
{"level":"error","message":"Failed to getPermissions:  Pipeline's scmHost github.com does not match with user's scmHost undefined","stack":"Error: Pipeline's scmHost github.com does not match with user's scmHost undefined\n    at GithubScm.lookupScmUri (/Users/tkyi/repos/opensource/screwdriver/node_modules/screwdriver-scm-github/index.js:231:27)\n    at GithubScm._getPermissions (/Users/tkyi/repos/opensource/screwdriver/node_modules/screwdriver-scm-github/index.js:865:40)\n    at /Users/tkyi/repos/opensource/screwdriver/node_modules/screwdriver-scm-base/index.js:355:25","timestamp":"2021-11-17T00:32:35.390Z"}
```

Reply:
OK, I will add a PR.